analysis_driver/client.py

```diff
@@ -98,6 +105,24 @@ def _process_dataset(d):
         (AsanaNotification, d, cfg.query('notification', 'asana'))
     )
 
+    def _handle_exception(exception):
+        import traceback
+        stacktrace = traceback.format_exc()
+        app_logger.critical('Encountered a %s exception: %s', exception.__class__.__name__, str(exception))
+        app_logger.info('Stack trace below:\n' + stacktrace)
+        _handle_termination(9)
+        ntf.crash_report(stacktrace)
```

Comment:
We should call crash_report in _handle_termination.
We need the the notification to be triggered (Asana or email) and the stacktrace will be useful to know when/where the process was stopped.